src/nni_manager/training_service/dlts/dltsTrainingService.ts

```diff
@@ -0,0 +1,473 @@
+        await restServer.start();
+        restServer.setEnableVersionCheck = this.versionCheck;
+        this.log.info(`DLTS Training service rest server listening on: ${restServer.endPoint}`);
+        await Promise.all([
+            this.statusCheckingLoop(),
+            this.submitJobLoop()]);
+        this.log.info('DLTS training service exit.');
+    }
+
+    private async statusCheckingLoop(): Promise<void> {
+        while (!this.stopping) {
+            const updateDLTSTrialJobs: Promise<void>[] = [];
+            for (const [trialJobId, dltsTrialJob] of this.trialJobsMap) {
+                updateDLTSTrialJobs.push(this.getDLTSTrialJobInfo(dltsTrialJob));
+            }
+    
+            await Promise.all(updateDLTSTrialJobs);
+
+            const restServer: DLTSJobRestServer = component.get(DLTSJobRestServer);
+            if (restServer.getErrorMessage !== undefined) {
+                throw new Error(restServer.getErrorMessage);
+            }
+            await delay(3000);
+        }
+    }
+
+    private async getDLTSTrialJobInfo(dltsTrialJob: DLTSTrialJobDetail): Promise<void> {
+        if (this.dltsClusterConfig == null) {
+            throw Error('Cluster config is not set');
+        }
+        const requestOptions: request.Options = {
+            uri: `${this.dltsClusterConfig.dashboard}api/v2/clusters/${this.dltsClusterConfig.cluster}/jobs/${dltsTrialJob.dltsJobId}`,
+            qs: {
+                email: this.dltsClusterConfig.email,
+                password: this.dltsClusterConfig.password
+            },
+            json: true
+        };
+        const body = await new Promise((resolve, reject) => {
+            request(requestOptions, (error, response, body) => {
+                if (error != null) {
+                    reject(error)
+                } else {
+                    resolve(body)
+                }
+            })
+        }) as any;
+        switch (body['jobStatus']) {
+            case 'unapproved':
+            case 'queued':
+            case 'scheduling':
+                dltsTrialJob.status = "WAITING";
+                break;
+            case 'running':
+                dltsTrialJob.status = "RUNNING";
+                if (dltsTrialJob.startTime === undefined) {
+                    dltsTrialJob.startTime = Date.parse(body['jobStatusDetail'][0]['startedAt'])
+                }
+                if (dltsTrialJob.url === undefined) {
+                    dltsTrialJob.url = `${this.dltsClusterConfig.dashboard}job/${this.dltsClusterConfig.team}/${this.dltsClusterConfig.cluster}/${dltsTrialJob.dltsJobId}`
+                }
+                break;
+            case 'finished':
+                dltsTrialJob.status = "SUCCEEDED";
+                break;
+            case 'failed':
+                dltsTrialJob.status = "FAILED";
+                break;
+            case 'pausing':
+                dltsTrialJob.status = "RUNNING";
+                dltsTrialJob.status = "RUNNING";
```

Comment:
duplicated line

Reply:
In DLTS job could be temporary "paused" (stop, release resources but easy to restart). I have not found a more suitable NNI status here.

Reply:
@Gerhut just curious, how users restart a paused job? the job starts from the beginning or could automatically resume the state?

Reply:
Starts from beginning, user should handle checkpoints by themselves.

Reply:
You could add a new status in NNI, I think `RUNNING` is not suitable here, since the job is not running.

Reply:
then the restart is triggered by user or by system. if it is triggered by user, better to show "paused" state. if triggered by system when there is available resource, then "running" might be fine.

Reply:
Triggered by user, there is a "resume" action to paused job.

Reply:
Can I make it `WAITING`?

Reply:
I think it is different from 'waiting'. if we show this state, users will be confused: why a job becomes waiting again after running.

Reply:
I'm not sure how much affection if I introduced a new state into NNI? At least shall I provide a new style of the `PAUSED` state to the NNI dashboard?

Reply:
as we discussed, if a job is in `PAUSED` state, simply cancel it.

Reply:
Done